piccolo_api/crud/endpoints.py

```diff
@@ -871,7 +889,16 @@ async def detail(self, request: Request) -> Response:
         try:
             row_id = self.table._meta.primary_key.value_type(row_id)
         except ValueError:
-            return Response("The ID is invalid", status_code=400)
+            for i in self.table._meta._foreign_key_references:
+                target = i._meta.params["target_column"]
```

Comment:
This should be `i._meta.params.get("target_column")` in case the key doesn't exist.